crypto/ml_kem/ml_kem.c

```diff
@@ -1524,13 +1526,15 @@ int decap(uint8_t secret[ML_KEM_SHARED_SECRET_BYTES],
         || !encrypt_cpa(tmp_ctext, decrypted, r, tmp, mdctx, key)) {
         memcpy(secret, failure_key, ML_KEM_SHARED_SECRET_BYTES);
         OPENSSL_cleanse(decrypted, ML_KEM_SHARED_SECRET_BYTES);
+        OPENSSL_cleanse(failure_key, ML_KEM_RANDOM_BYTES);
         return 1;
     }
     mask = constant_time_eq_int_8(0,
         CRYPTO_memcmp(ctext, tmp_ctext, vinfo->ctext_bytes));
     for (i = 0; i < ML_KEM_SHARED_SECRET_BYTES; i++)
         secret[i] = constant_time_select_8(mask, Kr[i], failure_key[i]);
     OPENSSL_cleanse(decrypted, ML_KEM_SHARED_SECRET_BYTES);
+    OPENSSL_cleanse(failure_key, ML_KEM_RANDOM_BYTES);
```

Comment:
Cleaning something that was never used is pointless.

I suggest you wrapper this and most of the others like this:
```suggestion
#ifdef OPENSSL_PEDANTIC_ZEROIZATION
    OPENSSL_cleanse(failure_key, ML_KEM_RANDOM_BYTES);
#endif
```